content/client-lib-development-guide/features.textile

```diff
@@ -267,7 +268,20 @@ h3(#rest-channel). Channel
+*** @(RSL6c1)@ Following the procedure described in "RTL4c":#RTL4c under the caveats of "RTL4d":#RTL4d, "RTL4e":#RTL4e and "RTL4f":#RTL4f. The @ProtocolMessage.channelSerial@ of the @ATTACH@ @ProtocolMessage@ sent as per "RTL4c":#RTL4c  should be set to the channel serial of the previous message to the message for which "vcdiff" decoding failed.
+*** @(RSL6c2)@ Any subsequent "vcdiff" decoding failures as per "RSL6c":#RSL6c while the recovery procedure from a previous failure as per "RSL6c1":#RSL6c1 is still in progress, i.e. the channel is in @ATTACHING@ state, should not trigger a new recovery procedure as per "RSL6c1":#RSL6c1.
+** @(RSL6d)@ The data payload of the last message on each channel, i.e. @Message.data@ has to be stored at all times since it might be needed to decode the "vcdiff" encoding. The last message bookkeeping rules are:
+*** @(RSL6d1)@ If the message encoding string ends in @base64@, the message should be base64-decoded (and the encoding string modified accordingly per "RSL6":#RSL6) before storing the last message
+*** @(RSL6d2)@ If "vcdiff" encoding is present the direct output of decoding the "vcdiff" encoding is stored as a last message, no matter that more encodings might be present
+** @(RSL6e)@ The @id@ of the last received message on each channel has to be stored. In case the next message received is delta message (i.e. contains @vcdiff@ encoding) the stored last message id has to be compared against the @Message.extras.delta.from@ of the delta message. If these are different the delta application fails and the recovery procedure from "RSL6c":#RSL6c must be executed.
+
+h3(#plugins). Plugins
+* @(PC1)@ Specific client library features that are not commonly used could be extracted to their own libraries as plugins in order to reduce the size of the client library. While residing in their own library such plugins are considered to be logical part of the client library code and as such are tightly coupled with the client library implementation. The client library is  aware of the plugin specific type and capabilities. I.e. each such plugin is by design coupled to a specific version of the client library.
+* @(PC2)@ The plugins are not expected to share any common methods. Yet there exists the convenience interface @Plugin@ that should be used in the strongly typed languages as the type of the @ClientOptions.plugins@ collection as per "TO3o":#TO3o.
+* @(PC3)@ A plugin provided under the type @PluginType.vcdiff@ should be capable of decoding "vcdiff" encoded messages. It must implement the the @VCDiffDecoder@ interface and the client library must be able to use it by casting it to this interface.
```

Comment:
>  A plugin provided under the type @PluginType.vcdiff@

I had to search around for a bit to understand what you meant here because I assumed that was a type, whereas it's an enum value.  So perhaps:

`A plugin provided with the @PluginType@ enum key value of @vcdiff@`